Lowercase tokens

Thaniela opened the pull request:
hello fixed the capitalization of the token names array :>

Pipluppp commented:
Would also have to update the names of the tokens used across scanner.c and parser.c

Pipluppp commented:
Need to change the token enum since that's what actually contains the tokens.

Thaniela commented:
The changes are working in scanner.c but not in parser.c

The ff are the changes I made
token.h
- lowercased the token array names, token enum, and keywordState
- renamed keyword tokens with added '_kw" to avoid token names conflicting with C language reserved words

scanner.c 
- renamed the functions: identifier > identifier_function (scanner.c)
- renamed the function: string > string_function (scanner.c)
- renamed the function: character_literal > character_literal_function (scanner.c)
- updated the tokens used 

parser.c 
- updated the tokens used

PaullyMac commented:
It seems like the updated code raised by @Thaniela works on my end, with no errors on my test run (using the test file of @rn-mnsl [i.e., `test_power_exp.core` and our main test file [i.e., `test_parse.core`]. The compilation of the updated `scanner.c` also works  (after all, I wouldn't be able to test the `parser.c` if not for that LOL!).

Here are some snippets upon executing the code:

1) ![image](https://github.com/user-attachments/assets/47da7ca9-04d0-4158-9f6c-61dd4bd74d9c)


2) changes on the left side showed that all of the token names used are now in lowercase letters
![image](https://github.com/user-attachments/assets/54131e5b-f2d8-41b3-ad6a-f356b3f375d9)

Thanks and have a great evening!

Pipluppp commented:
Goods naman yung tests, thank you!
